x/fungible/keeper/evm.go

```diff
@@ -825,5 +821,9 @@ func (k Keeper) CallEVMWithData(
 		}
 	}
 
+	if res.Failed() {
+		return res, cosmoserrors.Wrap(evmtypes.ErrVMExecution, fmt.Sprintf("%s: ret 0x%x", res.VmError, res.Ret))
```

Comment:
Avoid the usage of fmt.Sprintf when possible. It's an expensive function and we have Wrapf.
```suggestion
		return res, cosmoserrors.Wrapf(evmtypes.ErrVMExecution, "%s: ret 0x%x", res.VmError", res.Ret)
```